sparkles/roll_optimize.py

```diff
@@ -353,6 +353,7 @@ def improve_metric(n_stars, P2, n_stars_new, P2_new):
                             del kwargs[key]
 
             kwargs['att'] = q_att_roll
+            kwargs['raise_exc'] = True
```

Comment:
Remind me why we want this again?

Reply:
Now that it will be the default, not really necessary, but it doesn't hurt.  The need for this was highlighted by the issue yesterday with getting roll options with include_ids set. Debugging this was more difficult because the exception got swallowed and what finally ended up breaking was obscure and unrelated to the root problem.

Reply:
I just don't understand why we want to raise the exception when this will be called by the MATLAB tools and I don't think the exception is handled in our code.

Reply:
Specifically, on the MATLAB side, we can set raise_exc to False (or whatever @jskrist wants) in get_aca_catalog, but if raise_exc is explicitly set to True in the roll catalogs fetching with no controlling option when called from MATLAB, wouldn't there need to be new code to handle exceptions in the MATLAB Python?

Reply:
Have you checked the sparkles tidy?  https://github.com/sot/sparkles/pull/71

Reply:
Ah, you mean the new code to add an option to control this is in the next PR?  I can deal with that.